compiler/record-minmax/src/RecordMinMax.cpp

```diff
@@ -209,6 +215,28 @@ void RecordMinMax::initialize(const std::string &input_model_path)
   _interpreter->attachObserver(_observer.get());
 }
 
+void RecordMinMax::initialize_with_parallel_record(const std::string &input_model_path)
```

Comment:
How about not introduce this function? And we can only use introducing vector containers like `_interpreters` and `observers` instead of using existing `_interpreter` and `_oberver`. Then, codes will be look like below.
```cxx
// We could introduce dedicated context variable instead
void RecordMinMax::initialize(const std::string &input_model_path, bool in_parallel) 
// void RecordMinMax::initialize(const RMMConext ctx)
{
  initializeCircleModule(input_model_path);

  // Create and initialize interpreters and observers
  const auto threads_size = in_parallel ? omp_get_max_threads() : 1;

  _interpreters.resize(threads_size);
  _observers.resize(threads_size);

  for (uint32_t thread_idx = 0; thread_idx < threads_size; ++thread_idx)
  {
    auto interpreter = std::make_unique<luci_interpreter::Interpreter>(_module.get());
    auto observer = std::make_unique<MinMaxObserver>();

    interpreter->attachObserver(observer.get());

    _observers[thread_idx] = std::move(observer);
    _interpreters[thread_idx] = std::move(interpreter);
  }
}
```

Reply:
done